### PR TITLE
Decaffeinate client

### PR DESCRIPTION
**Pull request by srallen**
This removes coffeescript using an automated method using [decaffeinate](https://github.com/decaffeinate/decaffeinate). I also followed a couple of the output suggestions and noted which ones. Others will take some time. The tests appear to be passing.

I also removed all the browserify related dev dependencies. I don't think they were actually being used. The CLI build command used coffeescript's built in compiler.

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -1,182 +1,154 @@
...
+ * DS103: Rewrite code to no longer use __guard__, or convert again using --optional-chaining (note: done by srallen)
+ * DS205: Consider reworking code to avoid use of IIFEs
+ * DS206: Consider reworking classes to avoid initClass
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
+class SugarClient {
+  static initClass() {
+    this.host = null;
+    this.Primus = null;
```
Comment (mcbouslog) [inline client.js:14]: Running `npm link` and testing with PFE hit an issue `Uncaught TypeError: Cannot set property 'Primus' of undefined` here. I think related to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#binding_this_with_prototype_and_static_methods, I'm making some progress troubleshooting, but want to post update asap.

Comment (srallen) [reply client.js:14]: @mcbouslog thanks for checking. I'll continue this into the week. It's more important that survey task is finished than this gets decaffeinated.

**Review by mcbouslog — APPROVED** [review]
LGTM, tests passing, code looks good. 

Had the same issues trying `npm link` with this branch and master, unsure of why at the moment.

**Comment by srallen** [comment]
@mcbouslog I've spent a bit of time today trying to understand what is going on with npm link here. I checked out the commit before I updated coffeescript and it didn't work there either. I had a look at more specifically how sugar-client is being used in PFE and found that the sugar-client dependency isn't used directly by PFE at all despite it being included in the in the dependencies in the package.json. The client that is used is bundled with the panoptes-javascript-client. 

In the PJC is where the sugar client is instantiated with primus: https://github.com/zooniverse/panoptes-javascript-client/blob/master/lib/sugar.js

Now here's the problem with npm link. This function expects you to symlink by package name. I'm not sure if we _can_ symlink the primus file for how it's required in PJC. We could attempt to alias it via webpack, but I what I did instead is just copy and pasted the updated outputted compiled javascript file (on master) into the node_modules folder for the sugar client and that worked rather than use npm link. I'm going to guess that npm link has never worked with how this is setup.

This is putting aside the fact that the sugar client is including a dependency (primus.js) as a file rather than installed via package.json or declared a peer dependency. Maybe this wasn't an option six years ago....

**Comment by srallen** [comment]
@mcbouslog if you have time, could you re-review this using the copy/paste strategy I used and let me know if it works on PFE?

**Review by mcbouslog — APPROVED** [review]
I copy/pasted the decaffeinated `client.js` in PFE `node_modules` and sugar-client looks good (checked some notifications). Thanks for doing this!

**Pull request merged**